pagopa/io-app
# chore: [IA-144] Refactored the pin creation screen

## Short description
This PR is going to introduce a refactor to the PIN creation screen used during the onboarding and in the profile section.

<details>
<summary>iOS Preview</summary>

https://user-images.githubusercontent.com/5963683/167444733-c87b14e3-1079-482e-9242-e12f6bc9ae0e.mov

</details>

<details>
<summary>Android Preview</summary>

https://user-images.githubusercontent.com/5963683/167445805-97e08a2c-a706-4f90-9ad1-c9e326ad6de5.mov

</details>

✅ The E2E tests successfully ran on https://github.com/pagopa/io-app/pull/3939/commits/7e2a156304b63d4fc85b34cedc838a0ae2082630

## List of changes proposed in this pull request
- Created an agnostic `PinCreationForm` component (this will be useful in the future to remove the `isOnboardingCompleted` behaviour separating everything in two different screens, one for the onboarding and one for the profile)
- Replaced the old view in `PinScreen` with the new `PinCreationForm` in order to obtain the desired experience
- Updated `@testing-library/react-native` to version `8.0.0` in order to use `await` for `findByX` and `waitFor` during the unit tests

## How to test
The pin creation screen during the onboarding and in the profile (`Profile` -> `Security` -> `Change the unlock code`) should match the Figma file in the story and should be able to set the pin correctly.
